flow/layers/physical_model_layer.h

```diff
@@ -39,88 +50,11 @@ class PhysicalModelLayer : public ContainerLayer {
 #endif  // defined(OS_FUCHSIA)
 
  private:
-  std::unique_ptr<PhysicalLayerShape> shape_;
   float elevation_;
   SkColor color_;
   SkScalar device_pixel_ratio_;
-};
-
-// Common interface for the shape operations needed by PhysicalModelLayer.
-//
-// Once Scenic supports specifying physical layers with paths we can get rid
-// of this class and the subclasses, and have a single implementation of
-// PhysicalModelLayer that holds an SkPath.
-// TODO(amirh): remove this once Scenic supports arbitrary shaped layers.
-class PhysicalLayerShape {
- public:
-  virtual ~PhysicalLayerShape() = default;
-  virtual const SkRect& getBounds() const = 0;
-  virtual SkPath getPath() const = 0;
-  virtual void clipCanvas(SkCanvas& canvas) const = 0;
-  virtual bool isRect() const = 0;
-#if defined(OS_FUCHSIA)
-  virtual const SkRRect& getFrameRRect() const = 0;
-#endif  // defined(OS_FUCHSIA)
-};
-
-class PhysicalLayerRRect final : public PhysicalLayerShape {
- public:
-  PhysicalLayerRRect(const SkRRect& rrect) { rrect_ = rrect; }
-
-  // |flow::PhysicalLayerShape|
-  const SkRect& getBounds() const override { return rrect_.getBounds(); }
-
-  // |flow::PhysicalLayerShape|
-  SkPath getPath() const override;
-
-  // |flow::PhysicalLayerShape|
-  void clipCanvas(SkCanvas& canvas) const override {
-    canvas.clipRRect(rrect_, true);
-  }
-
-  // |flow::PhysicalLayerShape|
-  bool isRect() const override { return rrect_.isRect(); }
-
-#if defined(OS_FUCHSIA)
-  // |flow::PhysicalLayerShape|
-  const SkRRect& getFrameRRect() const override { return rrect_; }
-#endif  // defined(OS_FUCHSIA)
-
- private:
-  SkRRect rrect_;
-};
-
-class PhysicalLayerPath final : public PhysicalLayerShape {
- public:
-  PhysicalLayerPath(const SkPath& path) {
-    path_ = path;
-#if defined(OS_FUCHSIA)
-    frameRRect_ = SkRRect::MakeRect(path.getBounds());
-#endif  // defined(OS_FUCHSIA)
-  }
-
-  // |flow::PhysicalLayerShape|
-  const SkRect& getBounds() const override { return path_.getBounds(); }
-
-  // |flow::PhysicalLayerShape|
-  SkPath getPath() const override { return path_; }
-
-  // |flow::PhysicalLayerShape|
-  void clipCanvas(SkCanvas& canvas) const override {
-    canvas.clipPath(path_, true);
-  }
-
-  // |flow::PhysicalLayerShape|
-  bool isRect() const override { return false; }
-
-#if defined(OS_FUCHSIA)
-  // Scenic does not currently support compositing arbitrary shaped layers,
-  // so we just use the path's bounding rectangle.
-  // |flow::PhysicalLayerShape|
-  const SkRRect& getFrameRRect() const override { return frameRRect_; }
-#endif  // defined(OS_FUCHSIA)
- private:
   SkPath path_;
+  bool isRect_;
```

Comment:
add a constructor initializer for isRect_

Reply:
done